boot/bootstate20.go

```diff
@@ -385,15 +392,20 @@ func (ks20 *bootState20Kernel) loadBootenv() error {
 
 	// find the bootloader and ensure it's an extracted run kernel image
 	// bootloader
-	opts := &bootloader.Options{
-		// we want extracted run kernel images for uc20
-		// TODO:UC20: the name of this flag is now confusing, as it is being
-		//            slightly abused to tell the uboot bootloader to just look
-		//            in a different directory, even when we don't have an
-		//            actual extracted kernel image for that impl
-		ExtractedRunKernelImage: true,
-	}
-	bl, err := bootloader.Find("", opts)
+
+	var opts *bootloader.Options
+	if ks20.blOpts != nil {
+		opts = ks20.blOpts
+	} else {
+		opts = &bootloader.Options{}
+	}
+	// we want extracted run kernel images for uc20
+	// TODO:UC20: the name of this flag is now confusing, as it is being
+	//            slightly abused to tell the uboot bootloader to just look
+	//            in a different directory, even when we don't have an
+	//            actual extracted kernel image for that impl
+	opts.ExtractedRunKernelImage = true
```

Comment:
this section is a bit odd I'm not sure what the right thing to do here though is, should we enforce that blOpts always has ExtractedRunKernelImage set to true?

Reply:
afaict this should be done only in the else above?  also probably time to find the different name and use it in a follow up. Maybe UC20RunMode ?

Reply:
hmm this is a weird situation. We could actually drop defining this flag for all blOpts passed to the function that are not nil as you suggest, and only set ExtractedRunKernelImage for the else case when we weren't provided a blOpts to this function, but that is a bit confusing, because currently actually the uboot implementation treats NoSlashBoot as really (NoSlashBoot | ExtractedRunKernelImage), so when InitramfsRunModeSelectSnapsToMount creates the blOpts as just NoSlashBoot, it just happens to work for uboot, but if InitramfsRunModeSelectSnapsToMount specifies NoSlashBoot and ExtractedRunKernelImage then grub works from the Initramfs, but would otherwise behave differently in userspace. So I guess all this is to say I'm really confused what the flag should be renamed to because it has conflicting usages across the bootloaders it seems.

Anyways, I'm happy to just specify ExtractedRunKernelImage in the else case here as that seems to work fine, and then clean this up more in a followup but I don't know what to do about it really.